src/hotspot/share/memory/arena.hpp

```diff
@@ -86,7 +86,7 @@ class Chunk: CHeapObj<mtChunk> {
 
 //------------------------------Arena------------------------------------------
 // Fast allocation of memory
-class Arena : public CHeapObj<mtNone> {
+class Arena {
```

Comment:
The allocation base class tells the reader that `Arena` has special allocation constraints, by removing this it is not apparent there are any such constraints. Can we at least add a comment?

I confess I've forgotten how not having an allocation base class prevents use of `new Foo()` ??

Reply:
We have a link-time check in debug builds that fails the build if someone tries to use the global new operators.